sdks/python/container/piputil.go

```diff
@@ -148,8 +148,11 @@ func findBeamSdkWhl(files []string, acceptableWhlSpecs []string) string {
 // SDK from source tarball provided in sdkSrcFile.
 func installSdk(files []string, workDir string, sdkSrcFile string, acceptableWhlSpecs []string, required bool) error {
 	sdkWhlFile := findBeamSdkWhl(files, acceptableWhlSpecs)
+
 	if sdkWhlFile != "" {
-		err := pipInstallPackage(files, workDir, sdkWhlFile, false, false, []string{"gcp"})
+		// by default, pip rejects to install wheel if same version already installed
+		isDev := strings.Contains(sdkWhlFile, ".dev")
```

Comment:
is this the problem for just wheels or also if tar.gz is passed ?

Reply:
tarball is not affected. checked the log if passing a tarball (same pipeline options above expect sdk_location)

```
DEBUG 2023-05-26T21:15:34.103230283Z Found artifact: dataflow_python_sdk.tar
INFO 2023-05-26T21:15:34.103325167Z 2023/05/26 21:15:34 Installing setup packages ...
INFO 2023-05-26T21:15:34.645905671Z Processing /var/opt/google/staged/dataflow_python_sdk.tar
INFO 2023-05-26T21:15:35.031355177Z Preparing metadata (setup.py): started
...
INFO 2023-05-26T21:16:30.151314981Z Successfully built apache-beam
INFO 2023-05-26T21:16:31.978811453Z Installing collected packages: apache-beam
INFO 2023-05-26T21:16:31.979188478Z Attempting uninstall: apache-beam
INFO 2023-05-26T21:16:31.989685770Z Found existing installation: apache-beam 2.49.0.dev0
...
INFO 2023-05-26T21:16:37.762490639Z Successfully installed apache-beam-2.49.0.dev0
DEBUG 2023-05-26T21:16:38.099652894Z Executing Python (worker sdk-0-0_sibling_1): python -m apache_beam.runners.worker.sdk_worker_main
DEBUG 2023-05-26T21:16:38.268671515Z Executing Python (worker sdk-0-0): python -m apache_beam.runners.worker.sdk_worker_main
```

btw now install from tarball costs 1 minute with sibling worker enabled (previously took 3.5 min when every cpu core run a container)

though it seems PostCommit test still not run with sibling worker

Reply:
So yes it's just for wheels

Reply:
> though it seems PostCommit test still not run with sibling worker

hmmm... batch tests running on more than 1 core machines, should use sibling workers. if you have a job id, I can take a look why it's not used.

Reply:
I see, I might have checked streaming pipeline (which is the case for #26908 that discovered 

checked a batch python postcommit job it indeed run on sibling worker